.github/workflows/integration.yml

```diff
@@ -0,0 +1,197 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Integration
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "adbc.h"
+      - "go/**"
+      - "c/**"
+      - "glib/**"
+      - "python/**"
+      - ".github/workflows/integration.yml"
+  push:
+    paths:
+      - "adbc.h"
+      - "go/**"
+      - "c/**"
+      - "glib/**"
+      - "python/**"
+      - ".github/workflows/integration.yml"
+
+concurrency:
+  group: ${{ github.repository }}-${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  cpp-conda-unix:
+    name: "Conda/${{ matrix.os }}"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: ["macos-latest", "ubuntu-latest"]    
+    env:
+      # Required for macOS
+      # https://conda-forge.org/docs/maintainer/knowledge_base.html#newer-c-features-with-old-sdk
+      CXXFLAGS: "-D_LIBCPP_DISABLE_AVAILABILITY"
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+      - name: Get Date
+        id: get-date
+        shell: bash
+        run: |
+          echo "::set-output name=today::$(/bin/date -u '+%Y%m%d')"
+      - name: Cache Conda
+        uses: actions/cache@v3
+        env:
+          # Increment this to reset cache manually
+          CACHE_NUMBER: 0
+        with:
+          path: ~/conda_pkgs_dir
+          key: conda-${{ runner.os }}-${{ steps.get-date.outputs.today }}-${{ env.CACHE_NUMBER }}-${{ hashFiles('ci/**') }}
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+          use-mamba: true
+      - name: Install Dependencies
+        shell: bash -l {0}
+        run: |
+          mamba install -c conda-forge \
+            'arrow-c-glib>=10.0.0' \
+            --file ci/conda_env_cpp.txt \
+            --file ci/conda_env_docs.txt \
+            --file ci/conda_env_glib.txt \
+            --file ci/conda_env_python.txt
+      - name: Build and Install (No ASan)
+        id: build-install
+        shell: bash -l {0}
+        run: |
+          # Python and others need something that don't use the ASAN runtime
+          rm -rf "$(pwd)/build"
+          export BUILD_ALL=1
+          export ADBC_BUILD_TESTS=OFF
+          export ADBC_USE_ASAN=OFF
+          export ADBC_USE_UBSAN=OFF
+          ./ci/scripts/cpp_build.sh "$(pwd)" "$(pwd)/build" "$HOME/local"
+          echo "CACHEPATH=$HOME/local" >> $GITHUB_OUTPUT
```

Comment:
What we could do here is add a small CMake project that just builds the tests and links to the driver manager, using an env var to choose what driver to load.

Reply:
well, i wanted to build all the drivers here so that i can have them accessible for the tests, and then do precisely what you said.